p2p/dandelion/src/tests/mod.rs

```diff
@@ -0,0 +1,130 @@
+mod router;
+mod txpool;
+
+use std::{collections::HashMap, future::Future, hash::Hash, sync::Arc};
+
+use futures::TryStreamExt;
+use tokio::sync::mpsc::{self, UnboundedReceiver};
+use tower::{
+    discover::{Discover, ServiceList},
+    util::service_fn,
+    Service, ServiceExt,
+};
+
+use crate::{
+    traits::{TxStoreRequest, TxStoreResponse},
+    State,
+};
+
+pub fn mock_discover_svc<Req: Send + 'static>() -> (
+    impl Discover<
+        Key = usize,
+        Service = impl Service<
+            Req,
+            Future = impl Future<Output = Result<(), tower::BoxError>> + Send + 'static,
+            Error = tower::BoxError,
+        > + Send
+                      + 'static,
+        Error = tower::BoxError,
+    >,
+    UnboundedReceiver<(u64, Req)>,
```

Comment:
Type aliases?

Reply:
This is test code so I think this is fine.

Reply:
can't do impl trait in type aliases